Sources/Turf/GeoJSON.swift

```diff
@@ -0,0 +1,411 @@
+        try container.encode(properties, forKey: .properties)
+        try container.encodeIfPresent(identifier, forKey: .identifier)
+    }
+}
+
+public struct FeatureCollection: GeoJSONObject {
+    public var identifier: FeatureIdentifier<Int, String>?
+    public var features: Array<GeoJSONObject> = []
+    public var properties: [String : AnyJSONType]?
+    
+    private enum CodingKeys: String, CodingKey {
+        case properties
+        case features
+    }
+    
+    private enum GeometryCodingKeys: String, CodingKey {
+        case type
+    }
+    
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        var features: Array<GeoJSONObject> = []
+        var featureTypes: Array<GeoJSONObject.Type> = []
+        
+        if var unkeyedContainer = try? container.nestedUnkeyedContainer(forKey: .features) {
+            while (!unkeyedContainer.isAtEnd) {
+                let feature = try unkeyedContainer.decode(Feature.self)
+
+                if let test = feature.simplifiedGeometry?.geometryType {
+                    switch test {
+                    case .LineString:
+                        featureTypes.append(LineStringFeature.self)
+                    case .MultiLineString:
+                        featureTypes.append(MultiLineStringFeature.self)
+                    case .Point:
+                        featureTypes.append(PointFeature.self)
+                    case .Polygon:
+                        featureTypes.append(PolygonFeature.self)
+                    case .MultiPolygon:
+                        featureTypes.append(MultiPolygonFeature.self)
+                    case .MultiPoint:
+                        featureTypes.append(MultiPointFeature.self)
+                    }
+                }
+            }
+        }
+        
+        if var mappedContainer = try? container.nestedUnkeyedContainer(forKey: .features) {
+            while (!mappedContainer.isAtEnd) {
+                let featureType = featureTypes[mappedContainer.currentIndex]
+                if featureType is LineStringFeature.Type {
+                    features.append(try mappedContainer.decode(LineStringFeature.self))
+                } else if featureType is PolygonFeature.Type {
+                    features.append(try mappedContainer.decode(PolygonFeature.self))
+                } else if featureType is PointFeature.Type {
+                    features.append(try mappedContainer.decode(PointFeature.self))
+                } else if featureType is MultiPoint.Type {
+                    features.append(try mappedContainer.decode(MultiPointFeature.self))
+                } else if featureType is MultiLineString.Type {
+                    features.append(try mappedContainer.decode(MultiLineStringFeature.self))
+                } else if featureType is MultiPolygon.Type {
+                    features.append(try mappedContainer.decode(MultiPolygonFeature.self))
```

Comment:
This is pretty repetitive. Could we use `type(of:)` to consolidate all these cases into one case?

Reply:
Fixed in https://github.com/mapbox/turf-swift/pull/45/commits/c3edfbee5d0c2ab0c731009f2f9f518c2e0d8e53